crates/sui-framework/tests/delegation_tests.move

```diff
@@ -21,6 +22,93 @@ module sui::delegation_tests {
     const DELEGATOR_ADDR_1: address = @0x42;
     const DELEGATOR_ADDR_2: address = @0x43;
 
+    #[test]
+    fun test_split_join_staked_sui() {
+        let scenario_val = test_scenario::begin(DELEGATOR_ADDR_1);
+        let scenario = &mut scenario_val;
+        // All this is just to generate a dummy StakedSui object to split and join later
```

Comment:
You can also add a `#[test-only]` function in `staking_pool` to create dummy StakedSui objects with whatever fields you want